File: src/main/kotlin/inflearn/java/_2_1_큰_수_출력하기.java
package inflearn.java;

import java.util.ArrayList;
import java.util.Scanner;
import java.util.StringTokenizer;

public class _2_1_큰_수_출력하기 {
    public static int[] solution(int[] ints) {
        ArrayList<Integer> answer = new ArrayList<>();

        int prev = Integer.MIN_VALUE;
        for (int anInt : ints) {
            if(anInt > prev) {
                answer.add(anInt);
            }
            prev = anInt;
        }
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = Integer.parseInt(sc.nextLine());
        int[] ints = new int[n];

        StringTokenizer st = new StringTokenizer(sc.nextLine());
        for (int i = 0; i < n; i++) {
            ints[i] = Integer.parseInt(st.nextToken());
        }
        ints = solution(ints);

        for (int anInt : ints) {
            System.out.print(anInt + " ");
        }
    }
}

File: src/main/kotlin/inflearn/2_1 큰 수 출력하기.kt
package inflearn

import java.util.*

object `2_1 큰 수 출력하기` {
    fun solution(ints: IntArray): IntArray {
        val answer = ArrayList<Int>()
        var prev = Int.MIN_VALUE
        for (anInt in ints) {
            if (anInt > prev) {
                answer.add(anInt)
            }
            prev = anInt
        }
        return answer.toIntArray()
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val n = sc.nextLine().toInt()
        var ints = IntArray(n)
        val st = StringTokenizer(sc.nextLine())
        for (i in 0 until n) {
            ints[i] = st.nextToken().toInt()
        }
        ints = solution(ints)
        for (anInt in ints) {
            print("$anInt ")
        }
    }
}